packages/mask/src/extension/popups/pages/Wallet/index.tsx

```diff
@@ -101,16 +103,22 @@ export default function Wallet() {
     }, [location.search, location.pathname, chainId, TransactionFormatter])
 
     useEffect(() => {
-        if (!(isLocked && !getLockStatusLoading && !exclusionDetectLocked.some((x) => x === location.pathname))) return
-        navigate(urlcat(PopupRoutes.Unlock, { from: location.pathname }), { replace: true })
-    }, [isLocked, location.pathname, getLockStatusLoading])
+        if (getLockStatusLoading) return
+        if (isLocked) navigate(urlcat(PopupRoutes.Unlock, { from: location.pathname }), { replace: true })
+    }, [isLocked, getLockStatusLoading, location.pathname])
 
     useEffect(() => {
         return CrossIsolationMessages.events.requestsUpdated.on(({ hasRequest }) => {
             if (hasRequest) retry()
         })
     }, [retry])
 
+    useEffect(() => {
+        if (!wallet || loadingHasPassword || isLocked || getLockStatusLoading) return
+
+        navigate(hasPassword ? PopupRoutes.Wallet : PopupRoutes.SetPaymentPassword, { replace: true })
```

Comment:
Merge it into the previous `useEffect` that contains `navigate` either. And please avoid navigating in `useEffect`. It causes a flash of blank pages.